mayastor/src/target/iscsi.rs

```diff
@@ -82,101 +88,70 @@ thread_local! {
     static ADDRESS: RefCell<Option<String>> = RefCell::new(None);
 }
 
-/// Generate iqn based on provided uuid
-fn target_name(uuid: &str) -> String {
-    format!("iqn.2019-05.io.openebs:{}", uuid)
+/// Generate iqn based on provided bdev_name
+pub fn target_name(bdev_name: &str) -> String {
+    format!("iqn.2019-05.io.openebs:{}", bdev_name)
 }
 
 /// Create iscsi portal and initiator group which will be used later when
 /// creating iscsi targets.
 pub fn init(address: &str) -> Result<()> {
-    let portal_host = CString::new(address.to_owned()).unwrap();
-    let portal_port = CString::new(ISCSI_PORT.to_string()).unwrap();
-    let initiator_host = CString::new("ANY").unwrap();
-    let initiator_netmask = CString::new("ANY").unwrap();
 
-    let pg = unsafe { spdk_iscsi_portal_grp_create(0) };
-    if pg.is_null() {
-        return Err(Error::CreatePortalGroup {});
+    if let Err(e) = create_portal_group(address, ISCSI_PORT_BE, ISCSI_PORTAL_GROUP_BE) {
```

Comment:
you can simply write:

```rust
create_portal_group(...)?;
```

If there is an error, it will return it.

Reply:
Ok, will change as suggested.

Reply:
Fixed in commit https://github.com/openebs/Mayastor/commit/8fe37c12e6ad29c1c6c7d94da22f9d67a47a026b